### PR TITLE
Some updates in Horizon fixing some issues for upcoming release

### PR DESCRIPTION
**Pull request by Garyfallidis**
Images 

- [x] Automatic contrast adjustment on load
- [x] 3D and 4D image support
- [x] Slider for seeing threshold 
- [x] Provides access to colormaps
- [x] Picking voxels and showing underlying value
- [x] World or native coordinates supported

Streamlines
- [x] Uses distinguished_colormap to load multiple tractograms when ``--random_colors`` is set
- [x] Native coordinates disabled for now until this is updated with the StatefulTractogram
- [x] On the fly clustering with ``--cluster --cluster_thr 15`` continues being supported
- [x] Slider for changing clustering threshold

Overall
- [x]  Added global memory to communicate across actors and UI components.
- [ ] Increase coverage

### CODE REVIEW

File: dipy/viz/app.py [hunk 1]
```diff
@@ -197,8 +210,122 @@ def build_scene(self):
...
+        def left_click_cluster_callback(obj, event):
+
+            if self.cla[obj]['selected']:
+                self.cla[obj]['centroid_actor'].VisibilityOn()
+                ca = self.cla[obj]['centroid_actor']
+                self.cea[ca]['selected'] = 0
+                obj.VisibilityOff()
+                self.cea[ca]['expanded'] = 0
+
+            show_m.render()
```
Comment (skoudoro) [inline dipy/viz/app.py]: missing a self here?

File: dipy/viz/app.py [hunk 2]
```diff
@@ -509,9 +701,9 @@ def horizon(tractograms, images, cluster, cluster_thr, random_colors,
         adaptive visualization, Proceedings of: International Society of
         Magnetic Resonance in Medicine (ISMRM), Montreal, Canada, 2019.
     """
-    hz = Horizon(tractograms, images, cluster, cluster_thr, random_colors,
-                 length_lt, length_gt, clusters_lt, clusters_gt,
-                 world_coords, interactive, out_png)
+    hz = Horizon(tractograms, images, pams, cluster, cluster_thr, random_colors,
```
Comment (skoudoro) [inline dipy/viz/app.py]: pep8: `random_colors` too long

File: dipy/viz/panel.py [hunk 3]
```diff
@@ -31,35 +33,92 @@ def build_label(text, font_size=18, bold=False):
...
+            value_range = (0, 1.)
+            HORIZON.slicer_rgb = True
+    if ndim == 3:
+        value_range = np.percentile(tmp, q=[2, 98])
+
     if not world_coords:
-        image_actor_z = actor.slicer(data, affine=np.eye(4))
+        affine = np.eye(4)
+
+    # renderer.add(actor.axes(scale=(50, 50, 50)))
```
Comment (skoudoro) [inline dipy/viz/panel.py]: remove comment

File: dipy/viz/panel.py [hunk 4]
```diff
@@ -83,74 +142,367 @@ def slicer_panel(renderer, data=None, affine=None, world_coords=False):
...
+
     line_slider_x = ui.LineSlider2D(min_value=0,
                                     max_value=shape[0] - 1,
                                     initial_value=shape[0] / 2,
                                     text_template="{value:.0f}",
                                     length=140)
 
+    _color_slider(line_slider_x)
+
+
```
Comment (skoudoro) [inline dipy/viz/panel.py]: pep8: remove empty line

File: dipy/viz/gmem.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    slicer_grid = False
+
+    # tractogram level sharing
+    cluster_thr = 15
+    streamline_actors = []
+    centroid_actors = []
+    cluster_actors = []
+
+
+HORIZON = GlobalHorizon()
```
Comment (skoudoro) [inline dipy/viz/gmem.py]: HORIZON to HORIZONMEM

**Comment by Garyfallidis** [comment]
I suggest merging these fixes for now. More to come with 1.1 including increasing coverage for Horizon. I was planning to increase coverage here but there is an issue with xvfb which needs more investigation. Nonetheless, we have not seen any unexpected issues with the desktop version of this in all 3 operating systems (windows, linux and osx). So, I suggest to merge and write an issue to follow up.

**Comment by skoudoro** [comment]
Thank you @Garyfallidis

**Pull request merged**